### PR TITLE
ref(segmentedControl): Update styles

### PR DESCRIPTION
**Pull request by vuluongj20**
Update `SegmentedControl`'s animation, shadow, and container display mode. See comments below for details.

Part of https://github.com/getsentry/sentry/issues/43865

### CODE REVIEW

File: static/app/components/segmentedControl.tsx [hunk 1]
```diff
@@ -323,6 +324,7 @@ const VisibleLabel = styled('span')<{
   font-weight: ${p => (p.isSelected ? 600 : 400)};
   letter-spacing: ${p => (p.isSelected ? '-0.015em' : 'inherit')};
   text-align: center;
+  line-height: ${p => p.theme.text.lineHeightBody};
```
Comment (vuluongj20) [inline static/app/components/segmentedControl.tsx:327]: Slipping this in…

**Before ——** descenders (e.g. hook at the bottom of the letter g) are cropped
Screenshot 2023-02-01 at 10 55 53 AM

**After ——**
Screenshot 2023-02-01 at 10 55 19 AM

File: static/app/components/segmentedControl.tsx [hunk 2]
```diff
@@ -153,7 +153,8 @@ function Segment({
 
 const GroupWrap = styled('div')<{priority: Priority; size: FormSize}>`
   position: relative;
-  display: inline-flex;
+  display: inline-grid;
+  grid-auto-flow: column;
```
Comment (vuluongj20) [inline static/app/components/segmentedControl.tsx:157]: Use grid display to automatically stretch segments out to fill the outer container, if the container is wider than its contents.

**Before ——**
Screenshot 2023-02-01 at 10 03 59 AM

**After ——**
Screenshot 2023-02-01 at 10 03 40 AM

This only applies when `SegmentedControl`'s width is being controlled by either restyling or being inside a flex/grid container. By default the container will "hug" its contents:
Screenshot 2023-02-01 at 10 06 14 AM

File: static/app/components/segmentedControl.tsx [hunk 3]
```diff
@@ -245,7 +246,7 @@ const SegmentSelectionIndicator = styled(motion.div)<{priority: Priority}>`
     p.priority === 'primary'
       ? p.theme.borderRadius
       : `calc(${p.theme.borderRadius} - 1px)`};
-  box-shadow: 0 0 2px rgba(43, 34, 51, 0.16);
+  box-shadow: 0 0 2px rgba(43, 34, 51, 0.32);
```
Comment (vuluongj20) [inline static/app/components/segmentedControl.tsx:249]: Heavier shadow to better separate the selected segment from the rest.

**Before ——**
Screenshot 2023-02-02 at 10 32 19 AM

**After ——**
Screenshot 2023-02-02 at 10 32 02 AM

File: static/app/components/segmentedControl.tsx [hunk 4]
```diff
@@ -129,7 +129,7 @@ function Segment({
       {isSelected && (
         <SegmentSelectionIndicator
           layoutId={layoutGroupId}
-          transition={{type: 'tween', ease: 'circOut', duration: 0.2}}
+          transition={{type: 'tween', ease: 'easeOut', duration: 0.2}}
```
Comment (vuluongj20) [inline static/app/components/segmentedControl.tsx:132]: Different easing curve with less of a sudden bounce, based on feedback from @JonasBa.

File: static/app/components/segmentedControl.tsx [hunk 5]
```diff
@@ -110,8 +110,8 @@ function Segment({
 
   const {inputProps} = useRadio({...props}, state, ref);
 
-  const prevOptionIsSelected = state.selectedValue === prevKey;
-  const nextOptionIsSelected = state.selectedValue === nextKey;
+  const prevOptionIsSelected = prevKey && state.selectedValue === prevKey;
+  const nextOptionIsSelected = nextKey && state.selectedValue === nextKey;
```
Comment (vuluongj20) [inline static/app/components/segmentedControl.tsx:114]: Fix a hvoer overflow issue when no segment is selected:

**Before ——**
Screenshot 2023-02-02 at 11 20 42 AM

**After ——**
Screenshot 2023-02-02 at 11 19 14 AM

File: static/app/components/segmentedControl.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/components/segmentedControl.tsx:328]: nice

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**